histocc/attacker.py

```diff
@@ -5,64 +5,121 @@
+            x_string_copy = [random.sample(x,1) for x in x_string_copy]
+                
+        else:
+            # Alter chars
+            for i in range(len(x_string_copy)):
+                # alt_prob probability that nothing will happen to the string
+                if random.random() < alt_prob:
+                    continue
 
-            string_i = x_string_copy[i]
+                string_i = x_string_copy[i]
 
-            num_letters = len(string_i)
-            num_replacements = int(num_letters * alt_prob)
+                num_letters = len(string_i)
+                num_replacements = int(num_letters * alt_prob)
 
-            indices_to_replace = random.sample(range(num_letters), num_replacements)
+                indices_to_replace = random.sample(range(num_letters), num_replacements)
 
-            # Convert string to list of characters
-            chars = list(string_i)
+                # Convert string to list of characters
+                chars = list(string_i)
 
-            for j in indices_to_replace:
-                chars[j] =  random.choice(string.ascii_lowercase) # replace with a random letter
+                for j in indices_to_replace:
+                    chars[j] = random.choice(string.ascii_lowercase)  # replace with a random letter
 
-            string_i = ''.join(chars)
+                string_i = ''.join(chars)
 
-            x_string_copy[i] = string_i
+                x_string_copy[i] = string_i
 
-        if insert_words:
-            for i in range(len(x_string_copy)):
-                if random.random() < alt_prob: # Only make this affect alt_prob of cases
-                    # Word list
-                    occ_as_word_list = x_string_copy[i].split()
+            if insert_words:
+                for i in range(len(x_string_copy)):
+                    if random.random() < alt_prob:  # Only make this affect alt_prob of cases
+                        # Word list
+                        occ_as_word_list = x_string_copy[i].split()
 
-                    # Random word
-                    random_word = random.choice(self.word_list)
+                        # Random word
+                        random_word = random.choice(self.word_list)
 
-                    # choose a random index to insert the word
-                    insert_index = random.randint(0, len(occ_as_word_list))
+                        # choose a random index to insert the word
+                        insert_index = random.randint(0, len(occ_as_word_list))
 
-                    # insert the word into the list
-                    occ_as_word_list.insert(insert_index, random_word)
+                        # insert the word into the list
+                        occ_as_word_list.insert(insert_index, random_word)
 
-                    x_string_copy[i] = " ".join(occ_as_word_list)
+                        x_string_copy[i] = " ".join(occ_as_word_list)
 
         return x_string_copy
+
+if __name__ == '__main__':
```

Comment:
While helpful when developing, I think we should remove this `if __name__ == '__main__`. This script should never be used as the entry point, but keeping it signals valid entrypoint.